### DISCUSSION TITLE
[FR]: Content hash in the filename with Webpack (and other bundlers)

### DISCUSSION THREAD

**Discussion by njlr** [aspect-build/rules_js#2199] [General]
### What is the current behavior?

I am unsure if this is possible

### Describe the feature

It is common to put the hash of bundle artefacts into the filename to simplify browser caching. It is not clear how to do this with Webpack being executed by `rules_js` since Bazel expects static `outs`. 

If this is already possible then an example would be very helpful. 

Thanks!

### Fund our work

- [ ] Sponsor our open source work by donating a [feature bounty](https://opencollective.com/aspect-build/)

**Comment by gregmagolan** [comment]
Bazel expects file outputs to have deterministic names but you can also configure rules with directory outputs where only the directory names need to be deterministic and the files contained within the directory can include hashes in their filenames.

Instead of `outs` use the `out_dirs` attribute and configure your webpack config accordingly: https://docs.aspect.build/rules/aspect_rules_js/docs/js_run_binary#out_dirs

Does that answer your question?
